homeassistant/components/androidtv/media_player.py

```diff
@@ -192,68 +210,98 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         else:
             device_name = "Android TV / Fire TV device"
 
-        _LOGGER.warning("Could not connect to %s at %s %s", device_name, host, adb_log)
+        _LOGGER.warning(
+            "Could not connect to %s at %s %s", device_name, address, adb_log
+        )
         raise PlatformNotReady
 
-    if host in hass.data[ANDROIDTV_DOMAIN]:
-        _LOGGER.warning("Platform already setup on %s, skipping", host)
+    if address in hass.data[ANDROIDTV_DOMAIN]:
+        _LOGGER.warning("Platform already setup on %s, skipping", address)
+        return
+
+    if aftv.DEVICE_CLASS == DEVICE_ANDROIDTV:
+        device = AndroidTVDevice(
```

Comment:
If the only difference between these two code paths is the class that's used, just store the class in a variable and the default name .

Reply:
I stored the arguments and reused them.  Is this OK?

https://github.com/home-assistant/home-assistant/blob/728b0341c29d2693b5516fcc51cd061843ceca5b/homeassistant/components/androidtv/media_player.py#L222-L236

Reply:
@balloob is this change OK? Is this pull request ready to be merged?